### PR TITLE
Feature/multiple jobs

### PR DESCRIPTION
**Pull request by iluhinsky**

### CODE REVIEW

File: perftracker/models/job.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ class JobModel(models.Model):
 
     project         = models.ForeignKey(ProjectModel, help_text="Job project", on_delete=models.CASCADE)
 
+    regression      = models.ForeignKey('perftracker.RegressionModel', help_text="Regression of this job", related_name="regr", null=True, blank=True, on_delete=models.CASCADE)
```
Comment (perfguru87) [inline perftracker/models/job.py]: I think ondelete=models.CASCADE is wrong. It will delete the job when regression is deleted. The right option is SET_NULL

File: perftracker/models/regression.py [hunk 2]
```diff
@@ -74,16 +75,18 @@ def ptOnJobSave(job):
 
         if r is None:
             regression_tag = job.regression_tag if job.regression_tag else "regression"
-            r = RegressionModel(title="%s regression" % job.project.name, tag=regression_tag, project=job.project, jobs=0)
+            r = RegressionModel(title="%s regression" % job.project.name, tag=regression_tag, project=job.project, jobs=1)
```
Comment (perfguru87) [inline perftracker/models/regression.py]: why 1?

File: perftracker/models/regression.py [hunk 3]
```diff
@@ -38,10 +39,10 @@ def save(self):
         super(RegressionModel, self).save()
 
     def ptGetJobs(self):
```
Comment (perfguru87) [inline perftracker/models/regression.py]: As I understand now you should have 2 methods:

ptGetAllJobs() - list all the jobs with given regression tag
ptGetLinkedJobs() - list the jobs included into regression (there can be jobs with given tag but manually excluded)

File: perftracker/models/job.py (hunk 1 above)
Comment (perfguru87) [inline perftracker/models/job.py]: Now it's not clear what is the difference between:
- regression
and
- regression_tag

one of the attributes is what was assigned on test start and the other is what user manually included/excluded on th portal.

What is what?

I'd propose to have it like this:

1. regression_tag - is an immutable attribute set by the test client. Technically there can be a tag to a removed regression.

2. regression - is a foreign key to the ModelRegression object, i.e. this is real link. This is managed by the user of the perftracker portal

File: perftracker/views.py [hunk 4]
```diff
@@ -493,11 +493,7 @@ class RegressionJson(BaseDatatableView):
         max_display_length = 5000
 
         def render_column(self, row, column):
-            # We want to render user as a custom column
-            if column == 'tests_total':
-                return '{0} {1}'.format(row.tests_total, row.tests_completed)
-            else:
-                return super(JobJson, self).render_column(row, column)
+            return super(RegressionJson, self).render_column(row, column)
```
Comment (perfguru87) [inline perftracker/views.py:531]: It could be separate bugfix patch

File: perftracker/models/regression.py [hunk 5]
```diff
@@ -20,6 +20,7 @@
 from perftracker.models.comparison import ptCmpChartType, ptCmpTableType
 from perftracker.helpers import pt_float2human, pt_cut_common_sfx
 
+
```
Comment (perfguru87) [inline perftracker/models/regression.py]: It can be separate patch

File: perftracker/models/job.py [hunk 6]
```diff
@@ -48,6 +48,8 @@ class JobModel(models.Model):
     regression      = models.ForeignKey('perftracker.RegressionModel', help_text="Regression of this job", related_name="regr", null=True, blank=True, on_delete=models.CASCADE)
     regression_tag  = models.CharField(max_length=512, help_text="MyProduct-2.0 regression", null=True, blank=True, db_index=True)
 
+    shown           = models.BooleanField(help_text="Show job in regression", db_index=True, default=True)
+
```
Comment (perfguru87) [inline perftracker/models/job.py]: Oh. Now I see you also have the  "shown" attribute.
1. The name is too vague. "shown" where?
2. Why is the 'regression' foreign key not enough?

File: perftracker/models/job.py [hunk 7]
```diff
@@ -219,7 +231,7 @@ class Meta:
         model = JobModel
         fields = ('id', 'title', 'suite_name', 'suite_ver', 'env_node', 'end', 'duration', 'upload',
                   'tests_total', 'tests_completed', 'tests_failed', 'tests_errors', 'tests_warnings', 'project',
-                  'product_name', 'product_ver')
+                  'product_name', 'product_ver', 'shown')
```
Comment (perfguru87) [inline perftracker/models/job.py]: Probably it's better to add here 'regression_tag' and 'regression'.

File: perftracker/models/job.py [hunk 8]
```diff
@@ -186,6 +188,16 @@ def ptGenFileName(self):
             name = "job-%d-%s" % (self.id, name)
         return re.sub(r'[^a-zA-Z0-9_-]', '_', name)
 
+    @staticmethod
+    def ptChangeShowStatus(job_id, new_show_status):
```
Comment (perfguru87) [inline perftracker/models/job.py]: Here I'd suppose to name it:
ptChangeRegressionLink()

File: perftracker/static/js/perftracker.js [hunk 9]
```diff
@@ -284,6 +284,13 @@ function pt_draw_regression_details(d, err_msg)
         }
         s += "<td>{0}</td><td>{1}</td><td><a href='/{2}/job/{3}'>{4}</a></td>".ptFormat(hw, job.suite_ver, job.project, job.id, job.title);
         s += "<td>{0}</td><td>{1} (of {2})</td><td>{3}</td>".ptFormat(job.duration, job.tests_completed, job.tests_total, job.tests_errors);
+
+        console.log(job.id + ": " + job.shown);
```
Comment (perfguru87) [inline perftracker/static/js/perftracker.js]: need to remove debug log. It's too specific

File: perftracker/static/js/perftracker.js [hunk 10]
```diff
@@ -284,6 +284,13 @@ function pt_draw_regression_details(d, err_msg)
...
+        var showEditIcon = job.shown
+            ? `<td><a href="#" data-id={0} data-toggle="modal" class="pt-show-edit pt-show-edit--shown glyphicon glyphicon-check"></a></td>`.ptFormat(job.id)
```
Comment (perfguru87) [inline perftracker/static/js/perftracker.js]: pt-show-edit name is too vague...

File: perftracker/templates/regression_all.html [hunk 11]
```diff
@@ -150,6 +150,32 @@
             });
         }
     });
+
```
Comment (perfguru87) [inline perftracker/templates/regression_all.html:153]: I'd add prefix as before the code above:

"3. Jobs to regressions linking"

File: perftracker/models/job.py (hunk 1 above)
Comment (perfguru87) [inline perftracker/models/job.py]: You forgot to add migrations (for addiging 'regression' key).

Also, it makes sense to write custom migration script where you link the jobs to existing Regressions...

**Review by perfguru87 — CHANGES_REQUESTED** [review]

File: perftracker/models/job.py (hunk 1 above)
Comment (iluhinsky) [reply perftracker/models/job.py]: Yes, it's true.

File: perftracker/models/regression.py (hunk 2 above)
Comment (iluhinsky) [reply perftracker/models/regression.py]: Because this method is called from JobModel.save() when we save job linked to this new regression. So, the number of jobs in this regression from the start is 1.

File: perftracker/views.py (hunk 4 above)
Comment (iluhinsky) [reply perftracker/views.py:531]: Got it

File: perftracker/static/js/perftracker.js (hunk 9 above)
Comment (iluhinsky) [reply perftracker/static/js/perftracker.js]: Got it

File: perftracker/models/job.py (hunk 1 above)
Comment (iluhinsky) [reply perftracker/models/job.py]: Understand

File: perftracker/static/js/perftracker.js (hunk 10 above)
Comment (iluhinsky) [reply perftracker/static/js/perftracker.js]: pt-show-job-edit is better?

File: perftracker/models/job.py (hunk 6 above)
Comment (iluhinsky) [reply perftracker/models/job.py]: Because of semantics.
"shown" field provide you ability to easy add or remove job from regression via the UI, because job is always linked to particular regression.
Actually, I don't know how to build convenient UX system of linking existed job to existed regression without admin privilege or special field like this.

File: perftracker/models/job.py (hunk 8 above)
Comment (iluhinsky) [reply perftracker/models/job.py]: Understand

File: perftracker/templates/regression_all.html (hunk 11 above)
Comment (Jenkslex) [reply perftracker/templates/regression_all.html]: Don't get it
